NewForce-at-Mountwest/bangazon-api-radvioli
# Db computer resource

# Description

Allows the user to access the resources for the Computer table of the BangazonAPI. Able to create, update, and delete a Computer from the database as well as the ability to Get a single Computer and Get a list of all Computers in the Database. This resource is needed by Bangazon for keeping track of Employee Computers and whether or not a Computer has been decomissioned.

## Related Ticket(s)
Ticket #8 - Allow developers to access the computers resource

## Steps to Test

1. git add and git commit whatever you're working on on your branch
2. git checkout master to switch to your local master branch
3. git fetch --all will give you access to your teammate's branches (if they've pushed them up)
4. git checkout db-computer-resource to switch to their branch
5. If you don't see the most recent version of their files, try git pull origin db-computer-resource

Once on the db-computer-resource branch, go to Visual Studio and run your server. After your browser opens up navigate from api/values to api/computers and ensure the proper information is displayed (each entry has a Purchase Date, Decomission Date (if null, it will return zeros in a DateTime format). Once that is tested, scroll down in the Solution Explorer to TestComputer.cs inside TestBangazonAPI and right-click on it. Then click on Run Tests. Make sure all tests return passing.

# Checklist:
- [ ✔️  ] I have commented my code, particularly in hard-to-understand areas
- [ ✔️ ] I have made corresponding changes to the documentation
- [ ✔️ ] My changes generate no new errors
- [ ✔️ ] I have added tests that prove my fix is effective or that my feature works
- [ ✔️ ] New and existing unit tests pass locally with my changes
